From 9d06f86b729030962fdad5ba1229b86dae192057 Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Wed, 30 Aug 2023 13:21:59 +0200
Subject: [PATCH 02/11] sbp1: Update PID settings

Change settings as requested by IBM.

- Update integral coefficient: -9.2 -> -5
- Update negative hysteresis: 5 -> 0
- Update proportional coefficient: -536 -> -500
- Update ILimitMax and OutLimitMax: 32000 -> 18000
- Update setpoints to higher limits for all groups
- Drop QSFP as the ADC temperature sensors should not be
  used to measure the QFSP temperature

Change-Id: I9b0d3cdc1c538b2d534225b3c5224e846a17cb30
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/sbp1_chassis.json | 124 ++++++++++++-------------------
 1 file changed, 48 insertions(+), 76 deletions(-)

diff --git a/configurations/sbp1_chassis.json b/configurations/sbp1_chassis.json
index 83fe374..ee362fc 100644
--- a/configurations/sbp1_chassis.json
+++ b/configurations/sbp1_chassis.json
@@ -1564,8 +1564,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "BCM0_TEMP Temperature",
@@ -1574,41 +1574,13 @@
                 "BCM3_TEMP Temperature"
             ],
             "Name": "PSU Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
-            "OutLimitMin": 2500,
-            "Outputs": [],
-            "PCoefficient": -536,
-            "PositiveHysteresis": 0,
-            "SetPoint": 55,
-            "SlewNeg": 0,
-            "SlewPos": 0,
-            "Type": "Pid",
-            "Zones": [
-                "PSU"
-            ]
-        },
-        {
-            "Class": "temp",
-            "FFGainCoefficient": 0,
-            "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
-            "ILimitMin": 2500,
-            "Inputs": [
-                "QSFP1 Temperature",
-                "QSFP2 Temperature",
-                "QSFP3 Temperature",
-                "QSFP4 Temperature"
-            ],
-            "Name": "QSFP Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 95,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1620,8 +1592,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "Die_CPU1",
@@ -1630,13 +1602,13 @@
                 "Die_CPU4"
             ],
             "Name": "CPU Temperature",
-            "NegativeHysteresis": 2,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 65,
+            "SetPoint": 85,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1648,8 +1620,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "PVCCFA_EHV_CPU0 Temperature",
@@ -1674,13 +1646,13 @@
                 "PVPP_HBM_CPU3 Temperature"
             ],
             "Name": "CPU VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 85,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1692,8 +1664,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "P1V2_NIC Temperature",
@@ -1703,13 +1675,13 @@
                 "PVCORE_NIC2 Temperature"
             ],
             "Name": "NIC VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1721,8 +1693,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU1",
@@ -1743,13 +1715,13 @@
                 "DIMM_H2_CPU1"
             ],
             "Name": "CPU1 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1761,8 +1733,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU2",
@@ -1783,13 +1755,13 @@
                 "DIMM_H2_CPU2"
             ],
             "Name": "CPU2 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1801,8 +1773,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU3",
@@ -1823,13 +1795,13 @@
                 "DIMM_H2_CPU3"
             ],
             "Name": "CPU3 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
@@ -1841,8 +1813,8 @@
             "Class": "temp",
             "FFGainCoefficient": 0,
             "FFOffCoefficient": 0,
-            "ICoefficient": -9.2,
-            "ILimitMax": 32000,
+            "ICoefficient": -5,
+            "ILimitMax": 18000,
             "ILimitMin": 2500,
             "Inputs": [
                 "DIMM_A1_CPU4",
@@ -1863,13 +1835,13 @@
                 "DIMM_H2_CPU4"
             ],
             "Name": "CPU4 VR Temperature",
-            "NegativeHysteresis": 5,
-            "OutLimitMax": 32000,
+            "NegativeHysteresis": 0,
+            "OutLimitMax": 18000,
             "OutLimitMin": 2500,
             "Outputs": [],
-            "PCoefficient": -536,
+            "PCoefficient": -500,
             "PositiveHysteresis": 0,
-            "SetPoint": 55,
+            "SetPoint": 60,
             "SlewNeg": 0,
             "SlewPos": 0,
             "Type": "Pid",
-- 
2.41.0

